src/Microsoft.Data/StringColumn.cs

```diff
@@ -0,0 +1,199 @@
+        {
+            var lastBuffer = _stringBuffers[_stringBuffers.Count - 1];
+            if (lastBuffer.Count == int.MaxValue)
+            {
+                lastBuffer = new List<string>();
+                _stringBuffers.Add(lastBuffer);
+            }
+            lastBuffer.Add(value);
+            Length++;
+        }
+        private int GetBufferIndexContainingRowIndex(ref long rowIndex)
+        {
+            if (rowIndex > Length)
+            {
+                throw new ArgumentOutOfRangeException(strings.ColumnIndexOutOfRange, nameof(rowIndex));
+            }
+            int curArrayIndex = 0;
+            int numBuffers = _stringBuffers.Count;
+            while (curArrayIndex < numBuffers && rowIndex > _stringBuffers[curArrayIndex].Count)
+            {
+                rowIndex -= _stringBuffers[curArrayIndex].Count;
+                curArrayIndex++;
+            }
+            return curArrayIndex;
+        }
+
+        public override object this[long rowIndex]
+        {
+            get
+            {
+                int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                return _stringBuffers[bufferIndex][(int)rowIndex];
+            }
+            set
+            {
+                if (value is string)
+                {
+                    int bufferIndex = GetBufferIndexContainingRowIndex(ref rowIndex);
+                    _stringBuffers[bufferIndex][(int)rowIndex] = (string)value;
+                }
+                else
+                {
+                    throw new ArgumentException("Expected value to be a string", nameof(value));
+                }
+            }
+        }
+
+        public override object this[long startIndex, int length]
+        {
+            get
+            {
+                var ret = new List<string>();
+                int bufferIndex = GetBufferIndexContainingRowIndex(ref startIndex);
+                while (ret.Count < length && bufferIndex < _stringBuffers.Count)
+                {
+                    for (int i = (int)startIndex; ret.Count < length && i < _stringBuffers[bufferIndex].Count; i++)
+                    {
+                        ret.Add(_stringBuffers[bufferIndex][i]);
+                    }
+                    bufferIndex++;
+                    startIndex = 0;
+                }
+                return ret;
+            }
+        }
+        public override BaseColumn Sort(bool ascending = true)
+        {
+            PrimitiveColumn<long> columnSortIndices = GetAscendingSortIndices() as PrimitiveColumn<long>;
+            return Clone(columnSortIndices, !ascending);
+        }
+
+        public override BaseColumn GetAscendingSortIndices()
+        {
+            _GetSortIndices(Comparer<string>.Default, out PrimitiveColumn<long> columnSortIndices);
+            return columnSortIndices;
+
+        }
+        private void _GetSortIndices(Comparer<string> comparer, out PrimitiveColumn<long> columnSortIndices)
```

Comment:
(nit) even private methods shouldn't have underscores in their names - especially at the beginning.

Reply:
Fixed. Had to change PrimitiveColumn.BinaryOperations.cs too.